cranelift-wasm/src/func_translator.rs

```diff
@@ -206,7 +206,9 @@ fn parse_function_body<FE: FuncEnvironment + ?Sized>(
     while !state.control_stack.is_empty() {
         builder.set_srcloc(cur_srcloc(&reader));
         let op = reader.read_operator()?;
-        translate_operator(op, builder, state, environ)?;
+        environ.before_translate_operator(&op, builder)?;
+        translate_operator(&op, builder, state, environ)?;
```

Comment:
`op` being used in `{before,after}_translate_operator` is what necessitates the bulk of this PR, by changing to `&op` here.